log/log.go

```diff
@@ -5,112 +5,121 @@ import (
 	"io"
 	"log"
 	"os"
-	//	"github.com/journeymidnight/yig/helper"
+	"runtime"
+	"strings"
 )
 
-// These flags define which text to prefix to each log entry generated by the Logger.
+type Level int
+
 const (
-	// Bits or'ed together to control what's printed.
-	// There is no control over the order they appear (the order listed
-	// here) or the format they present (as described in the comments).
-	// The prefix is followed by a colon only when Llongfile or Lshortfile
-	// is specified.
-	// For example, flags Ldate | Ltime (or LstdFlags) produce,
-	//	2009/01/23 01:23:23 message
-	// while flags Ldate | Ltime | Lmicroseconds | Llongfile produce,
-	//	2009/01/23 01:23:23.123123 /a/b/c/d.go:23: message
-	Ldate         = 1 << iota     // the date in the local time zone: 2009/01/23
-	Ltime                         // the time in the local time zone: 01:23:23
-	Lmicroseconds                 // microsecond resolution: 01:23:23.123123.  assumes Ltime.
-	Llongfile                     // full file name and line number: /a/b/c/d.go:23
-	Lshortfile                    // final file name element and line number: d.go:23. overrides Llongfile
-	LUTC                          // if Ldate or Ltime is set, use UTC rather than the local time zone
-	LstdFlags     = Ldate | Ltime // initial values for the standard logger
+	ErrorLevel Level = 0 // Errors should be properly handled
+	WarnLevel  Level = 1 // Errors could be ignored; messages might need noticed
+	InfoLevel  Level = 2 // Informational messages
```

Comment:
I want to add  DebugLevel and FatalLevel

Reply:
For `DebugLevel`:
- debug level logger would lead to very casual style logs, like `helper.Debugln("-----------old object version:", err)`
- production systems would not enable debug-level, if you think a message is useful, make it "info"
- for local tests, just use `fmt.Println` and remove them when commiting

For `FatalLevel`:
- just use `panic()` and collect them from stderr
- I think Go standard library's `log.Fatal` suprises many people at first because a log method terminates the program